pyro/poutine/subsample_messenger.py

```diff
@@ -128,3 +129,20 @@ def _process_message(self, msg):
                 with ignore_jit_warnings():
                     msg["scale"] = torch.tensor(msg["scale"])
         msg["scale"] = msg["scale"] * self.size / self.subsample_size
+
+    def _postprocess_message(self, msg):
+        if msg["type"] == "param":
+            event_dim = msg["kwargs"].get("event_dim")
+            if event_dim is not None:
+                dim = self.dim - event_dim
```

Comment:
You'll need to check `self.dim is not None` to handle the case of sequential plates